connectors/sources/azure_blob_storage.py

```diff
@@ -190,19 +191,38 @@ async def get_content(self, blob, timestamp=None, doit=None):
             document,
             filename,
             file_extension,
-            partial(self.blob_download_func, filename, blob["container"]),
+            partial(self.blob_download_func, filename, blob["container"], file_size),
         )
 
-    async def blob_download_func(self, blob_name, container_name):
-        async with BlobClient.from_connection_string(
-            conn_str=self.connection_string,
-            container_name=container_name,
-            blob_name=blob_name,
-            retry_total=self.retry_count,
-        ) as blob_client:
-            data = await blob_client.download_blob()
-            async for content in data.chunks():
-                yield content
+    def _get_container_client(self, container_name):
```

Comment:
Will you need to close these clients manually?

Reply:
Yes, I've added a `close` method for the same.